nsls2forge_utils/auto_tick.py

```diff
@@ -0,0 +1,627 @@
+import json
+from uuid import uuid4
+from subprocess import SubprocessError, CalledProcessError
+
+import github3
+import networkx as nx
+
+from conda_forge_tick.utils import (
+    frozen_to_json_friendly,
+    setup_logger,
+    eval_cmd,
+    dump_graph,
+    load_graph,
+    LazyJson
+)
+from conda_forge_tick.contexts import (
+    MigratorContext,
+    FeedstockContext,
+    MigratorSessionContext
+)
+from conda_forge_tick.migrators import (
+    Version,
+    PipMigrator,
+    MigrationYaml,
+    LicenseMigrator,
+    CondaForgeYAMLCleanup,
+    ExtraJinja2KeysCleanup,
+    Jinja2VarsCleanup,
+)
+from conda_forge_tick.auto_tick import (
+    migration_factory,
+    _compute_time_per_migrator,
+)
+from conda_forge_tick.status_report import write_version_migrator_status
+from conda_forge_tick.git_utils import is_github_api_limit_reached
+from conda_forge_tick.xonsh_utils import indir, env
+from conda_forge_tick.mamba_solver import is_recipe_solvable
+
+from .git_utils import (
+    get_repo,
+    push_repo
+)
+
+logger = logging.getLogger(__name__)
+
+PR_LIMIT = 5
+MAX_PR_LIMIT = 50
+
+MIGRATORS = [
+    Version(
+        pr_limit=PR_LIMIT * 2,
+        piggy_back_migrations=[
+            Jinja2VarsCleanup(),
+            PipMigrator(),
+            LicenseMigrator(),
+            CondaForgeYAMLCleanup(),
+            ExtraJinja2KeysCleanup(),
+        ],
+    ),
+]
+
+BOT_RERUN_LABEL = {
+    "name": "bot-rerun",
+    "color": "#191970",
+    "description": (
+        "Apply this label if you want the bot to retry "
+        "issuing a particular pull-request"
+    ),
+}
+
+
+def run(feedstock_ctx, migrator, protocol='ssh', pull_request=True,
+        rerender=True, fork=False, organization='nsls-ii-forge', **kwargs):
+    """
+    For a given feedstock and migration run the migration
+
+    Parameters
+    ----------
+    feedstock_ctx: FeedstockContext
+        The node attributes
+    migrator: Migrator instance
+        The migrator to run on the feedstock
+    protocol : str, optional
+        The git protocol to use, defaults to ``ssh``
+    pull_request : bool, optional
+        If true issue pull request, defaults to true
+    rerender : bool
+        Whether to rerender
+    fork : bool
+        If true create a fork, defaults to true
+    organization: str, optional
+        GitHub organization to get repo from
+    gh : github3.GitHub instance, optional
+        Object for communicating with GitHub, if None build from $USERNAME
+        and $PASSWORD, defaults to None
+    kwargs: dict
+        The key word arguments to pass to the migrator
+
+    Returns
+    -------
+    migrate_return: MigrationUidTypedDict
+        The migration return dict used for tracking finished migrations
+    pr_json: dict
+        The PR json object for recreating the PR as needed
+    """
+    # get the repo
+    migrator.attrs = feedstock_ctx.attrs
+
+    branch_name = migrator.remote_branch(feedstock_ctx) + "_h" + uuid4().hex[0:6]
```

Comment:
OK, now it's clear how the branch name is generated 😄 
```suggestion
    branch_name = migrator.remote_branch(feedstock_ctx) + "_h" + uuid4().hex[:6]
```